File: test/Integration/test-java/src/main/java/com/function/AddProductExtraColumns.java
package com.function;

import com.microsoft.azure.functions.HttpMethod;
import com.microsoft.azure.functions.HttpRequestMessage;
import com.microsoft.azure.functions.HttpResponseMessage;
import com.microsoft.azure.functions.HttpStatus;
import com.microsoft.azure.functions.OutputBinding;
import com.microsoft.azure.functions.annotation.AuthorizationLevel;
import com.microsoft.azure.functions.annotation.FunctionName;
import com.microsoft.azure.functions.annotation.HttpTrigger;
import com.microsoft.azure.functions.sql.annotation.SQLOutput;
import com.function.Common.ProductExtraColumns;

import java.util.Optional;

public class AddProductExtraColumns {
    @FunctionName("AddProductExtraColumns")
    public HttpResponseMessage run(
            @HttpTrigger(
                name = "req",
                methods = {HttpMethod.GET},
                authLevel = AuthorizationLevel.ANONYMOUS,
                route = "addproduct-extracolumns")
                HttpRequestMessage<Optional<String>> request,
            @SQLOutput(
                commandText = "dbo.Products",
                connectionStringSetting = "sqlConnectionString")
                OutputBinding<ProductExtraColumns> product) {

        ProductExtraColumns p = new ProductExtraColumns(0, "test", 0, 0, "test");
        product.setValue(p);

        return request.createResponseBuilder(HttpStatus.OK).header("Content-Type", "application/json").body(product).build();
    }
}

File: test-outofproc/AddProductExtraColumns.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

using DotnetIsolatedTests.Common;
using Microsoft.Azure.Functions.Worker;
using Microsoft.AspNetCore.Http;
using Microsoft.Azure.Functions.Worker.Extensions.Sql;

namespace DotnetIsolatedTests
{
    public static class AddProductExtraColumns
    {
        // This output binding should throw an Exception because the ProductExtraColumns object has 
        // two properties that do not exist as columns in the SQL table (ExtraInt and ExtraString).
        [Function("AddProductExtraColumns")]
        [SqlOutput("dbo.Products", "SqlConnectionString")]
        public static ProductExtraColumns Run(
            [HttpTrigger(AuthorizationLevel.Anonymous, "post", Route = "addproduct-extracolumns")]
            HttpRequest req)
        {
            var product = new ProductExtraColumns
            {
                Name = "test",
                ProductId = 1,
                Cost = 100,
                ExtraInt = 1,
                ExtraString = "test"
            };
            return product;
        }
    }
}
